### PR TITLE
refactor listbox

### PR DESCRIPTION
**Pull request by anatolzak**
Listbox Refactor:
- setting `activeTrigger` in trigger action instead of in `openMenu` and in `safeOnMount` and in `effect([open])`
- using early return in `effect([open])` for cleaner code

### CODE REVIEW

**Comment by huntabyte** [comment]
Same comments as #1055

File: src/lib/builders/listbox/create.ts [hunk 1]
```diff
@@ -292,6 +285,7 @@ export function createListbox<
 			} as const;
 		},
 		action: (node: HTMLElement): MeltActionReturn<ListboxEvents['trigger']> => {
+			activeTrigger.set(node);
```
Comment (TGlide) [inline src/lib/builders/listbox/create.ts:288]: It should reset on destroy

**Review by TGlide — COMMENTED** [review]
Thank you, just one small change needed!

File: src/lib/builders/listbox/create.ts (hunk 1 above)
Comment (anatolzak) [reply src/lib/builders/listbox/create.ts:286]: Done!

**Review by TGlide — APPROVED** [review]

**Pull request merged**